vendor/github.com/cilium/tetragon/pkg/k8s/apis/cilium.io/client/register_v1_crd.go

```diff
@@ -115,6 +124,19 @@ func createTPCRDs(clientset apiextensionsclient.Interface) error {
 	); err != nil {
 		return err
 	}
+
+	if !skipPodInfo {
+		isoCRD = GetPregeneratedCRD(v1alpha1.PICRDName)
+		if err := createUpdateCRD(
+			clientset,
+			v1alpha1.PICRDName,
+			constructV1CRD(v1alpha1.PIName, isoCRD),
+			newDefaultPoller(),
+		); err != nil {
+			return err
+		}
+	}
+
 	return nil
```

Comment:
I think it would be a better approach to not give up, and just try everything. The new errors.Join can help us with that.
I also think we can refactor the code to avoid repetition. So how about something like:

```
// createTPCRDs creates or updates the CRDs with the API server.
func createCRDs(clientset apiextensionsclient.Interface, skipPodInfo bool) error {
	doCreateCRD := func(crdName, name string) error {
		isoCRD := GetPregeneratedCRD(crdName)
		err := createUpdateCRD(
			clientset,
			crdName,
			constructV1CRD(name, isoCRD),
			newDefaultPoller())
		if err != nil {
			err = fmt.Errorf("failed to create %s: %w", crdName, err)
			return err
		}
		return nil
	}

	var ret error
	errors.Join(ret, doCreateCRD(v1alpha1.TPCRDName, v1lpha.TPName))
	errors.Join(ret, doCreateCRD(v1alpha1.TPNamespacedCRDName, v1alpha1.TPNamespacedName))
	if skipPodInfo {
		return ret
	}
	errors.Join(ret, doCreateCRD(v1alpha1.PICRDName, v1alpha1.PIName))
	return ret
}
```

Reply:
Sure I have pushed the changes, please review. 

I haven't squashed the commits yet, will do it when no more changes are needed.